### PR TITLE
Update _split_one_line and remove whitespace parameter

### PR DESCRIPTION
**Pull request by xvlaurent**
This is an attempt to update _split_one_line function to avoid usage of 'whitespace' parameter and approximation made for '_atom_site' CIF block.

Performance have to be approved with a CI run.

### CODE REVIEW

**Comment by padix-key** [comment]
Hi, thanks for writing this PR together. The benchmarks show that this variant is indeed slightly faster than than the regex-based one. However is still about 40% slower than the fast `expect_whitespace=False` hack. This makes parsing a structure without bonds about 20% slower (as shown above).

Still, I tend towards accepting this change, because it is much cleaner and technically gives the correct result, while currently it does not. If high performance is desired, I would recommend using BinaryCIF anyway. @biotite-dev has anyone a strong opinion about it?

**Comment by padix-key** [comment]
Fixes #673

**Comment by xvlaurent** [comment]
I tryed to reduce/optimize the code a bit more, but it seems there is no real gain...

**Comment by padix-key** [comment]
This is okay, a few other people (including me) already tried that unsuccessfully. After Rust code is introduced into the project (#688), this is probably the first function that is replaced by fast Rust routine.

The PDBx test is currently failing, as it does not expect a generator. Could you either revert to the non-generator variant or adapt the test?

**Comment by xvlaurent** [comment]
Finally, all tests are fine and branch is up to date!

File: tests/database/test_rcsb.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ def test_fetch(format, as_file_like):
     if format == "pdb":
         file = pdb.PDBFile.read(file_path_or_obj)
         pdb.get_structure(file)
-    elif format == "pdbx":
+    elif format == "cif":
```
Comment (padix-key) [inline tests/database/test_rcsb.py:34]: Good catch!

File: src/biotite/structure/io/pdbx/cif.py [hunk 2]
```diff
@@ -1062,29 +1040,27 @@ def _split_one_line(line):
...
-        if field[0] == field[-1] == "'" or field[0] == field[-1] == '"':
-            field = field[1:-1]
-        fields.append(field)
-    return fields
+        yield line[1:]
+    else:
+        # Loop over the line
+        while line:
+            # Strip leading whitespace(s)
+            striped_line = line.lstrip()
```
Comment (padix-key) [inline src/biotite/structure/io/pdbx/cif.py:1048]: Only a small typo

```suggestion
            stripped_line = line.lstrip()
```

File: src/biotite/structure/io/pdbx/cif.py [hunk 3]
```diff
@@ -1062,29 +1040,27 @@ def _split_one_line(line):
...
+            # Split the line on whitespace
+            word, _, line = striped_line.partition(" ")
```
Comment (padix-key) [inline src/biotite/structure/io/pdbx/cif.py:1050]:
```suggestion
            word, _, line = stripped_line.partition(" ")
```

File: src/biotite/structure/io/pdbx/cif.py [hunk 4]
```diff
@@ -1062,29 +1040,27 @@ def _split_one_line(line):
...
+            if word.startswith(("'", '"')):
+                # Set the separator to the quote found
+                separator = word[0]
+                # Handle the case of a quoted word without space
+                if word.endswith(separator) and len(word) > 1:
+                    # Yield the word without the opening and closing quotes
+                    yield word[1:-1]
+                    continue
+                # split the word on the separator
+                word, _, line = striped_line[1:].partition(separator)
```
Comment (padix-key) [inline src/biotite/structure/io/pdbx/cif.py:1061]:
```suggestion
                word, _, line = stripped_line[1:].partition(separator)
```

**Review by padix-key — CHANGES_REQUESTED** [review]
I only found this small typo. Otherwise the PR looks good to merge!

**Pull request merged**